org.eclipse.lsp4xml/src/test/java/org/eclipse/lsp4xml/services/XMLCompletionTest.java

```diff
@@ -19,9 +35,118 @@
 public class XMLCompletionTest {
 
 	private XMLLanguageService languageService;
+	private FormattingOptions sharedFormattingOptions = new FormattingOptions(4, false);
 
 	@Before
 	public void initializeLanguageService() {
 		languageService = new XMLLanguageService();
 	}
+
+	@Test
+	public void successfulEndTagCompletion() {
+		assertEndTagCompletion("<a>|", 3, "$0</a>");
+		assertEndTagCompletion("<a><b>|</a>", 6, "$0</b>");
+		assertEndTagCompletion("<a>   <b>|</a>", 9, "$0</b>");
+		assertEndTagCompletion("<a><b>|", 6, "$0</b>");
+	}
+
+	@Test
+	public void unneededEndTagCompletion() {
+		assertEndTagCompletion("<a><a>|</a>", 6, "$0</a>");
+		assertEndTagCompletion("<a><b>|</b></a>", 6, "$0</b>");

```

Comment:
could we avoid completing with `</b>` here, as that'd would unbalance the xml document?

Reply:
With this example, <a><a></a>, the parser considers the 2nd and 3rd as a complete element and the 1st tag as incomplete so its hard to determine with this current implementation which tag is incomplete.